File: backend/src/main/java/com/codementor/backend/service/FeaturedProjectService.java
package com.codementor.backend.service;

import com.codementor.backend.dto.FeaturedProjectResponse;
import com.codementor.backend.dto.FeaturedProjectsRequest;

import java.util.List;

public interface FeaturedProjectService {

    /**
     * Returns the user's featured projects.
     */
    List<FeaturedProjectResponse> getFeaturedProjects(
            String email
    );

    /**
     * Replaces the user's featured projects.
     */
    void updateFeaturedProjects(
            String email,
            FeaturedProjectsRequest request
    );

}

File: frontend/src/services/featuredProjectService.ts
import api from "./api";

import type {
  FeaturedProject,
  FeaturedProjectsRequest,
} from "../types/featuredProject";

/**
 * Get the user's featured projects.
 */
export async function getFeaturedProjects(): Promise<FeaturedProject[]> {
  const response = await api.get<FeaturedProject[]>(
    "/portfolio/featured-projects"
  );

  return response.data;
}

/**
 * Update the user's featured projects.
 */
export async function updateFeaturedProjects(
  request: FeaturedProjectsRequest
): Promise<void> {
  await api.put(
    "/portfolio/featured-projects",
    request
  );
}